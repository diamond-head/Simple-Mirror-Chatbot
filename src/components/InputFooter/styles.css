.input-footer__container {
  background-color: var(--white);
  width: 100%;
  border-top: 0.5px solid var(--text-gray-50);
  position: fixed;
  bottom: 0;
  left: 0;
  display: flex;
  justify-content: space-between;
}

.input-footer__container .send-button {
  margin-right: 16px;
}
