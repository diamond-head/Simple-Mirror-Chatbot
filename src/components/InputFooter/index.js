import React from 'react';
import ControlledInput from '../ControlledInput';

import SendSvg from '../../assets/images/send.svg';
import "./styles.css";

export default function InputFooter({
  onSendButtonClick,
  onValueChange,
  defaultMessageValue
}) {
  return (
    <div className="input-footer__container">
      <ControlledInput 
        name={'message-input-box'}
        placeholder={"Type your message..."}
        defaultValue={defaultMessageValue}
        onValueChange={onValueChange}
      />
      <img className="send-button" src={SendSvg} alt="send-button" onClick={onSendButtonClick} />
    </div>
  );
}