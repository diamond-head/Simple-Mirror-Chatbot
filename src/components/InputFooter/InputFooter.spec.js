import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import InputFooter from './index';
import SendSvg from '../../assets/images/send.svg'

describe("<InputFooter /> component", () => {
  it("InputFooter snapshot renders", () => {
    const component = renderer.create(
      <InputFooter 
        onSendButtonClick={() => {}} 
        onValueChange={() => {}} 
        defaultMessageValue={""} 
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render an input-footer image", () => {
    const wrapper = shallow(<InputFooter />);
    expect(wrapper.find('img').prop('src')).toEqual(SendSvg)
  });
});
