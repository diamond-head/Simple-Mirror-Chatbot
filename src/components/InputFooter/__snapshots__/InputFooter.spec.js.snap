// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFooter /> component InputFooter snapshot renders 1`] = `
<div
  className="input-footer__container"
>
  <div
    className="controlled-input__container"
  >
    <textarea
      className="controlled-input__textarea"
      name="message-input-box"
      onChange={[Function]}
      placeholder="Type your message..."
      rows={1}
      value=""
    />
  </div>
  <img
    alt="send-button"
    className="send-button"
    onClick={[Function]}
    src="test-file-stub"
  />
</div>
`;
